### ISSUE TITLE
Missing copy info %folder%

### ISSUE THREAD

**Issue by mseliger** [danny0838/webscrapbook#252]
The legacy ScrapBookX has the option to add the folder to the copy page info, where the file is stored: %FOLDER%.
This option is missing in WebScrapBook. Do you want to add the option in future or is it a problem to copy the folder (when using the backend server), where the file is stored?

**Comment by danny0838** [comment]
This needs further evaluation. As WebScrapBook supports a more flexible tree structure (by allowing any item be put under any multiple items), this can cause a significant performance issue to finding a folder path (it may be multiple paths and possibly infinite paths due to circular referencing).

**Comment by danny0838** [comment]
What do you mean "is it a problem to copy the folder where the file is stored"? You mean the virtual sidebar tree folder or the physical filesystem folder?

**Comment by mseliger** [comment]
Yes, the virtual sidebar tree when you use the backend server in firefox.

**Comment by danny0838** [comment]
Not quite sure about your question. Basically copying a folder item to other where in the tree (with recursively copying descendants or not) is not a problem. You can simply try for yourself.

**Comment by mseliger** [comment]
I mean the the format of copy info. Here is %folder% missing.
```
This determines the format of copy info.

Following “%<param>%” template syntaxes are available:
• %% - a literal “%”.
• %id% - item ID.
• %title% - item title.
• %comment% - item comment.
• %source% - item source URL.
• %url% - item URL.
• %create-#% - item create time.
• %modify-#% - item modify time.
```

**Comment by danny0838** [comment]
@mseliger  I know this. I was asking you what you mean to ask for "is it a problem to copy the folder where the file is stored"?

**Comment by mseliger** [comment]
I want to know if you can add the function -  i think it is very useful if you want to use the data for other programs. I'll copy the title, path to the source, path to the file, the folder, the id as date from ScrapBook to the E-Book-Organizer to Calibre, where my epub-, mobi- and pdf-files are stored. So i can search with one program - the Calibre program - for all my data. I use %folder% there as keywords for Calibre.

**Comment by danny0838** [comment]
With some evaluation it seems implementable, and can possibly be implemented in the next release. A problem we are considering is how to format the folder paths. Legacy ScrapBook X Copy Info uses `f1/f2/f3`, while we are considering whether there's a better way or some way to make it customizable...

**Comment by danny0838** [comment]
Implemented `%folder%` and `%path%` in 0.144.0.

**Comment by mseliger** [comment]
Thank you very much - it works well.
